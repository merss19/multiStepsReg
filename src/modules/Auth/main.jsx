"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var react_router_1 = require("react-router");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var CustomModal_1 = require("modules/CustomModal");
var SignUp_1 = require("modules/SignUp");
var storage_1 = require("tools/storage");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state, ownProps) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isAuth: selectors_1.selectIsAuth(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    user: selectors_1.selectUser(state)
}); };
var ProtectedRouteComponent = /** @class */ (function (_super) {
    __extends(ProtectedRouteComponent, _super);
    function ProtectedRouteComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProtectedRouteComponent.prototype.componentWillMount = function () {
        var _this = this;
        var checkAuth = this.props.checkAuth;
        checkAuth(function (data) { return _this.checkAuth(data); }, function () { return _this.leave(); });
    };
    ProtectedRouteComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching ||
            nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    ProtectedRouteComponent.prototype.leave = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('txId', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('packageType', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.checkAuth = function (data) {
        var _a = this.props, changeStep = _a.changeStep, history = _a.history;
        if (data.paidState !== 0) {
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
                history.replace('/signup');
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else if (!data.emailConfirmed) {
                return null;
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            changeStep(SignUp_1.Steps.two);
        }
        else {
            changeStep(SignUp_1.Steps.four);
        }
    };
    ProtectedRouteComponent.prototype.modalClick = function () {
        var _a = this.props, history = _a.history, toggleModal = _a.toggleModal;
        toggleModal(false);
        history.replace('/');
    };
    ProtectedRouteComponent.prototype.clickHandler = function () {
        var isError = this.props.isError;
        if (!isError) {
            return null;
        }
        return this.modalClick();
    };
    ProtectedRouteComponent.prototype.renderModal = function (modal) {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg, isError = _a.isError;
        return (<CustomModal_1.CustomModal modal={modal} resultText={errMsg} clickHandler={this.clickHandler.bind(this)} isLoader={isFetching}/>);
    };
    ProtectedRouteComponent.prototype.renderContent = function (props) {
        var _a = this.props, ProtectedComponent = _a.component, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<div>
        <ProtectedComponent {...props}/>
        {this.renderModal(CustomModal_1.ModalTypes.error)}
      </div>);
    };
    ProtectedRouteComponent.prototype.renderRoute = function (props) {
        var isAuth = this.props.isAuth;
        return (<div>
        {isAuth ?
            this.renderContent(props) :
            this.renderModal(CustomModal_1.ModalTypes.info)}
      </div>);
    };
    ProtectedRouteComponent.prototype.render = function () {
        var _this = this;
        var _a = this.props, ProtectedComponent = _a.component, isAuth = _a.isAuth, isError = _a.isError, isFetching = _a.isFetching, errMsg = _a.errMsg, rest = __rest(_a, ["component", "isAuth", "isError", "isFetching", "errMsg"]);
        return (<react_router_dom_1.Route {...rest} render={function (props) {
            return _this.renderRoute(props);
        }}/>);
    };
    return ProtectedRouteComponent;
}(React.Component));
exports.ProtectedRouteComponent = ProtectedRouteComponent;
exports.ProtectedRoute = react_redux_1.connect(mapStateToProps, {
    checkAuth: ducks_1.checkAuth,
    changeStep: SignUp_1.changeStep,
    toggleModal: CustomModal_1.toggleModal
})(ProtectedRouteComponent);
exports.default = react_router_1.withRouter(exports.ProtectedRoute);
