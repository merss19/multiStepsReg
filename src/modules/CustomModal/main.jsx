"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var ReactModal = require("react-modal");
var const_1 = require("./const");
var Button_1 = require("components/Button");
var Loader_1 = require("components/Loader");
var ducks_1 = require("./ducks");
var selectors_1 = require("./selectors");
var react_motion_1 = require("react-motion");
var style = {
    overlay: {
        backgroundColor: 'rgba(45, 43, 43, 0.85)',
        zIndex: '2'
    },
    content: {
        transform: 'translate3d(0%, -50%, 0px)',
        top: '50%',
        padding: '40px',
        maxHeight: '300px',
        maxWidth: '400px',
        overflow: 'none',
        borderRadius: '12px',
        margin: '0 auto',
        textAlign: 'center',
        bottom: 'auto'
    }
};
var mapStateToProps = function (state, ownProps) { return ({
    isOpen: selectors_1.selectIsOpen(state)
}); };
var CustomModalComponent = /** @class */ (function (_super) {
    __extends(CustomModalComponent, _super);
    function CustomModalComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CustomModalComponent.prototype.setBtn = function () {
        var modal = this.props.modal;
        var btn = Button_1.ButtonTypes.info;
        switch (modal) {
            case const_1.ModalTypes.success:
                btn = Button_1.ButtonTypes.success;
                break;
            case const_1.ModalTypes.error:
                btn = Button_1.ButtonTypes.error;
                break;
            case const_1.ModalTypes.info:
                btn = Button_1.ButtonTypes.info;
                break;
            default:
                btn = Button_1.ButtonTypes.info;
        }
        return btn;
    };
    CustomModalComponent.prototype.renderContent = function () {
        var _this = this;
        var _a = this.props, resultText = _a.resultText, isLoader = _a.isLoader;
        var btn = this.setBtn();
        if (isLoader) {
            return <Loader_1.Loader />;
        }
        else {
            return (<react_motion_1.Motion defaultStyle={{ opacity: 0 }} style={{ opacity: react_motion_1.spring(1, { stiffness: 70 }) }}>
          {function (interpolatedStyle) {
                return (<div className='modal' style={__assign({}, interpolatedStyle)}>
                  <h2 className='modal__title'>{resultText}</h2>
                  <Button_1.Button onClick={_this.closeModal.bind(_this)} styleBtn={btn}>
                    Продолжить
                  </Button_1.Button>
                </div>);
            }}
        </react_motion_1.Motion>);
        }
    };
    CustomModalComponent.prototype.closeModal = function () {
        var _a = this.props, toggleModal = _a.toggleModal, clickHandler = _a.clickHandler;
        toggleModal(false);
        if (!clickHandler) {
            return;
        }
        clickHandler();
    };
    CustomModalComponent.prototype.render = function () {
        var isOpen = this.props.isOpen;
        return (<div>
        <ReactModal isOpen={isOpen} style={style} contentLabel='Action Modal' onRequestClose={this.closeModal.bind(this)}>
          <div className='modal__content'>
            {this.renderContent()}
          </div>

        </ReactModal>
      </div>);
    };
    CustomModalComponent.defaultProps = {
        isLoader: false
    };
    return CustomModalComponent;
}(React.PureComponent));
exports.CustomModalComponent = CustomModalComponent;
exports.CustomModal = react_redux_1.connect(mapStateToProps, { toggleModal: ducks_1.toggleModal })(CustomModalComponent);
exports.default = exports.CustomModal;
