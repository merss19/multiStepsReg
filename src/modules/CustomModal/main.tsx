import * as React from 'react';
import {connect} from 'react-redux';
import * as ReactModal from 'react-modal';
import {ModalType} from './interfaces';
import {ModalTypes} from './const';
import {Button, ButtonType, ButtonTypes} from 'components/Button';
import {Loader} from 'components/Loader';
import {toggleModal} from './ducks';
import {selectIsOpen} from './selectors';
import {Motion, spring, presets} from 'react-motion';

const style = {
  overlay: {
    backgroundColor: 'rgba(45, 43, 43, 0.85)',
    zIndex: '2'
  },
  content: {
    transform: 'translate3d(0%, -50%, 0px)',
    top: '50%',
    padding: '40px',
    maxHeight: '300px',
    maxWidth: '400px',
    overflow: 'none',
    borderRadius: '12px',
    margin: '0 auto',
    textAlign: 'center',
    bottom: 'auto'
  }
};

interface OwnProps {
  resultText: string
  isLoader?: boolean
  modal?: ModalType
  clickHandler?: () => void
}

type ConnectedDispatch = {
  toggleModal(data: boolean): void
}

type ConnectedState = {
  isOpen: boolean
}

const mapStateToProps = (state: any, ownProps: OwnProps): ConnectedState => ({
  isOpen: selectIsOpen(state)
});

export class CustomModalComponent extends React.PureComponent<ConnectedState & OwnProps & ConnectedDispatch, {}> {

  static defaultProps = {
    isLoader: false
  };

  setBtn() {
    const {modal} = this.props;
    let btn: ButtonType = ButtonTypes.info;
    switch (modal) {
      case ModalTypes.success:
        btn = ButtonTypes.success;
        break;
      case ModalTypes.error:
        btn = ButtonTypes.error;
        break;
      case ModalTypes.info:
        btn = ButtonTypes.info;
        break;
      default:
        btn = ButtonTypes.info;
    }
    return btn;
  }

  renderContent() {
    const {resultText, isLoader} = this.props;
    const btn: ButtonType = this.setBtn();

    if (isLoader) {
      return <Loader/>;
    } else {
      return (
        <Motion
          defaultStyle={{opacity: 0}}
          style={{opacity: spring(1, {stiffness: 70})}}>
          {
            interpolatedStyle => {
              return (
                <div className='modal' style={{...interpolatedStyle}}>
                  <h2 className='modal__title'>{resultText}</h2>
                  <Button
                    onClick={this.closeModal.bind(this)}
                    styleBtn={btn}
                  >
                    Продолжить
                  </Button>
                </div>
              );
            }
          }
        </Motion>
      )
    }
  }

  closeModal() {
    const {toggleModal, clickHandler} = this.props;
    toggleModal(false);
    if (!clickHandler) {
      return;
    }
    clickHandler();
  }

  render() {
    const {isOpen} = this.props;
    return (
      <div>
        <ReactModal
          isOpen={isOpen}
          style={style}
          contentLabel='Action Modal'
          onRequestClose={this.closeModal.bind(this)}
        >
          <div className='modal__content'>
            {this.renderContent()}
          </div>

        </ReactModal>
      </div>
    );
  }
}

export const CustomModal = connect(mapStateToProps, {toggleModal})(CustomModalComponent);

export default CustomModal;