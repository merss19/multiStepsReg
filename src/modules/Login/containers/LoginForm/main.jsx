"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var InputText_1 = require("components/InputText");
var _1 = require("components/Button/");
var formValidations = require("tools/validations");
var LoginForm = /** @class */ (function (_super) {
    __extends(LoginForm, _super);
    function LoginForm() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LoginForm.prototype.render = function () {
        var _a = this.props, handleSubmit = _a.handleSubmit, onSubmit = _a.onSubmit;
        return (<div className='entry entry--min'>
        <div className='entry__inner'>

          <div className='entry__header'>
            <h2 className='entry__title text-center'>Вход в Личный кабинет</h2>
          </div>

          <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
            <redux_form_1.Field name='email' placeholder='Ваш e-mail' type='email' component={InputText_1.default}/>
            <redux_form_1.Field name='password' placeholder='Ваш пароль' type='password' component={InputText_1.default}/>

            <_1.Button type='submit' styleBtn={_1.ButtonTypes.info} wide={true}>
              Войти
            </_1.Button>

            <ul className='entry-nav mtb20'>
              <li className='entry-nav__item'>
                <react_router_dom_1.Link to='/signup'>Регистрация</react_router_dom_1.Link>
              </li>
              <li className='entry-nav__item'>
                <react_router_dom_1.Link to='/restore'>Забыли пароль?</react_router_dom_1.Link>
              </li>
            </ul>

          </form>
          {this.props.children}
        </div>
      </div>);
    };
    return LoginForm;
}(React.PureComponent));
exports.LoginForm = LoginForm;
exports.default = redux_form_1.reduxForm({
    form: 'LoginFormm',
    validate: formValidations.validateLogin
})(LoginForm);
