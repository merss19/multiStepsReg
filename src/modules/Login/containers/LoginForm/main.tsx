import * as React from 'react';
import {Field, FormProps, reduxForm} from 'redux-form';
import {Link} from 'react-router-dom';
import InputText from 'components/InputText';
import {Button, ButtonTypes} from 'components/Button/';
import * as formValidations from 'tools/validations';

interface LoginFormData {
  email: string
  password: string
}

interface OwnProps extends FormProps<LoginFormData, {}, {}> {
  onSubmit: (values: LoginFormData) => void
}

export class LoginForm extends React.PureComponent<OwnProps, {}> {
  render() {
    const {handleSubmit, onSubmit} = this.props;
    return (
      <div className='entry entry--min'>
        <div className='entry__inner'>

          <div className='entry__header'>
            <h2 className='entry__title text-center'>Вход в Личный кабинет</h2>
          </div>

          <form onSubmit={handleSubmit(onSubmit)} className='entry__box'>
            <Field name='email'
                   placeholder='Ваш e-mail'
                   type='email'
                   component={InputText}
            />
            <Field name='password'
                   placeholder='Ваш пароль'
                   type='password'
                   component={InputText}
            />

            <Button
              type='submit'
              styleBtn={ButtonTypes.info}
              wide={true}
            >
              Войти
            </Button>

            <ul className='entry-nav mtb20'>
              <li className='entry-nav__item'>
                <Link to='/signup'>Регистрация</Link>
              </li>
              <li className='entry-nav__item'>
                <Link to='/restore'>Забыли пароль?</Link>
              </li>
            </ul>

          </form>
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default reduxForm({
  form: 'LoginFormm',
  validate: formValidations.validateLogin
})(LoginForm);
