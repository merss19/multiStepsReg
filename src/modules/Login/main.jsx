"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var LoginForm_1 = require("./containers/LoginForm");
var ducks_1 = require("./ducks");
var react_router_1 = require("react-router");
var CustomModal_1 = require("../CustomModal");
var LayoutHOC_1 = require("components/HOC/LayoutHOC");
var storage_1 = require("tools/storage");
var SignUp_1 = require("../SignUp");
var selectors_1 = require("./selectors");
var mapStateToProps = function (state) { return ({
    isFetching: selectors_1.selectIsFetching(state),
    isLoad: selectors_1.selectIsLoad(state),
    isError: selectors_1.selectIsError(state),
    errMsg: selectors_1.selectErrMsg(state),
    profile: selectors_1.selectProfile(state)
}); };
var LoginComponent = /** @class */ (function (_super) {
    __extends(LoginComponent, _super);
    function LoginComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    LoginComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.isFetching !== this.props.isFetching || nextProps.isError !== this.props.isError) {
            var isOpen = nextProps.isFetching || nextProps.isError;
            this.props.toggleModal(isOpen);
        }
    };
    LoginComponent.prototype.onSubmit = function (data) {
        var _this = this;
        this.props.submitLogin(data, function (data) { return _this.submitCb(data); });
    };
    LoginComponent.prototype.submitCb = function (data) {
        var _a = this.props, changeStep = _a.changeStep, setToken = _a.setToken, history = _a.history;
        if (data && data.paidState !== 0) {
            setToken(data.authToken);
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            storage_1.storage.save('userProgram', data.program, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            if (data.paidState === 2) {
                changeStep(SignUp_1.Steps.four);
            }
            else if (data.isFirstEdit) {
                history.replace('/profile');
            }
            else {
                history.replace('/task');
            }
        }
        else if (!data.program) {
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(SignUp_1.Steps.two);
            history.replace('/signup');
        }
        else {
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(SignUp_1.Steps.three);
            history.replace('/signup');
        }
    };
    LoginComponent.prototype.setError = function () {
        this.props.setError(false);
    };
    LoginComponent.prototype.render = function () {
        var _a = this.props, isFetching = _a.isFetching, errMsg = _a.errMsg;
        return (<LoginForm_1.LoginForm onSubmit={this.onSubmit.bind(this)}>
        <CustomModal_1.CustomModal modal={CustomModal_1.ModalTypes.error} resultText={errMsg} isLoader={isFetching} clickHandler={this.setError.bind(this)}/>
      </LoginForm_1.LoginForm>);
    };
    return LoginComponent;
}(React.Component));
exports.LoginComponent = LoginComponent;
exports.Login = react_redux_1.connect(mapStateToProps, {
    submitLogin: ducks_1.submitLogin,
    changeStep: SignUp_1.changeStep,
    setToken: ducks_1.setToken,
    setError: ducks_1.setError,
    toggleModal: CustomModal_1.toggleModal
})(LoginComponent);
exports.default = LayoutHOC_1.LayoutHOC()(react_router_1.withRouter(exports.Login));
