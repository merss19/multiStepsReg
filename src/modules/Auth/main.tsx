import * as React from 'react';
import {connect} from 'react-redux';
import {RouteProps, RouteComponentProps, withRouter} from 'react-router';
import {Route} from 'react-router-dom';
import {checkAuth} from './ducks'
import {CustomModal, toggleModal, ModalTypes, ModalType} from 'modules/CustomModal';
import {changeStep, Steps} from 'modules/SignUp';
import {storage} from 'tools/storage';
import {
  selectIsFetching,
  selectIsLoad,
  selectUser,
  selectIsAuth,
  selectErrMsg,
  selectIsError
} from './selectors'

type ConnectedDispatch = {
  checkAuth(cb: (data?: any) => void, cbFail: () => void): void
  toggleModal(data: boolean): void
  changeStep: (data: Steps) => void
}

interface OwnProps extends RouteProps {
}

type ConnectedState = {
  isFetching: boolean
  isLoad: boolean
  isError: boolean
  isAuth: boolean
  errMsg: string
  user: any
}

const mapStateToProps = (state: any, ownProps: OwnProps): ConnectedState => ({
  isFetching: selectIsFetching(state),
  isLoad: selectIsLoad(state),
  isAuth: selectIsAuth(state),
  isError: selectIsError(state),
  errMsg: selectErrMsg(state),
  user: selectUser(state)
});
type Props = RouteComponentProps<string> & OwnProps & ConnectedDispatch & ConnectedState;

export class ProtectedRouteComponent extends React.Component<Props, {}> {

  componentWillMount() {
    const {checkAuth} = this.props;
    checkAuth((data) => this.checkAuth(data), () => this.leave());
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.isFetching !== this.props.isFetching ||
        nextProps.isError !== this.props.isError) {
      const isOpen: boolean = nextProps.isFetching || nextProps.isError;
      this.props.toggleModal(isOpen);
    }
  }

  leave() {
    const {history, toggleModal} = this.props;
    toggleModal(false);
    storage.remove('token', {path: '/'});
    storage.remove('txId', {path: '/'});
    storage.remove('program', {path: '/'});
    storage.remove('packageType', {path: '/'});
    storage.remove('promoName', {path: '/'});
    history.replace('/');
  }

  checkAuth(data: any) : void {
    const {changeStep, history} = this.props;
    if (data.paidState !== 0) {
      storage.save('userProgram', data.program, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      if (data.paidState === 2) {
        changeStep(Steps.four);
        history.replace('/signup');
      } else if (data.isFirstEdit) {
        history.replace('/profile');
      } else if (!data.emailConfirmed) {
        return null;
      } else {
        history.replace('/task');
      }
    } else if (!data.program) {
      changeStep(Steps.two);
    } else {
      changeStep(Steps.four);
    }
  }

  modalClick(): void {
    const {history, toggleModal} = this.props;
    toggleModal(false);
    history.replace('/');
  }

  clickHandler(){
    const {isError} = this.props
    if(!isError ){
      return null
    }
    return this.modalClick()
  }


  renderModal(modal:ModalType){
    const {isFetching, errMsg, isError} = this.props
    return(
      <CustomModal
        modal={modal}
        resultText={errMsg}
        clickHandler={this.clickHandler.bind(this)}
        isLoader={isFetching}
      />)
  }

  renderContent(props: any){
    const {component: ProtectedComponent, isFetching, errMsg} = this.props
    return(
      <div>
        <ProtectedComponent {...props}/>
        {this.renderModal(ModalTypes.error)}
      </div>)
  }

  renderRoute(props: any){
    const {isAuth} = this.props;
    return(
      <div>
        {isAuth ?
          this.renderContent(props) :
          this.renderModal(ModalTypes.info)
        }
      </div>)
  }

  render() {
    const {component: ProtectedComponent, isAuth, isError, isFetching, errMsg, ...rest} = this.props;
    return (
      <Route {...rest} render={props => {
        return this.renderRoute(props);
      }
      }/>
      )
  }
}


export const ProtectedRoute = connect(
  mapStateToProps,
  {
    checkAuth,
    changeStep,
    toggleModal
  })(ProtectedRouteComponent)


export default withRouter(ProtectedRoute);

