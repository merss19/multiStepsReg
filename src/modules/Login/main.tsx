import * as React from 'react';
import {connect} from 'react-redux';
import {LoginForm} from './containers/LoginForm';
import {submitLogin, setToken, setError} from './ducks';
import {LoginFormData} from './interfaces';
import {withRouter, RouteComponentProps} from 'react-router'
import {CustomModal, toggleModal, ModalTypes} from '../CustomModal';
import {LayoutHOC} from 'components/HOC/LayoutHOC';
import {storage} from 'tools/storage';
import {changeStep, Steps} from '../SignUp';


import {
  selectIsFetching,
  selectIsLoad,
  selectProfile,
  selectErrMsg,
  selectIsError
} from './selectors'

type ConnectedDispatch = {
  changeStep: (data: Steps) => void
  submitLogin(data: LoginFormData, cb: (data: any) => void): void
  toggleModal(data: boolean): void
  setToken(toket: string): void
  setError(data: boolean): any
}
type ConnectedState = {
  isFetching: boolean
  isLoad: boolean
  isError: boolean
  errMsg: string,
  profile: any
}
const mapStateToProps = (state: any): ConnectedState => ({
  isFetching: selectIsFetching(state),
  isLoad: selectIsLoad(state),
  isError: selectIsError(state),
  errMsg: selectErrMsg(state),
  profile: selectProfile(state)
});

type props = RouteComponentProps<string> & ConnectedDispatch & ConnectedState;

export class LoginComponent extends React.Component<props, any> {

  componentWillReceiveProps(nextProps: props) {
    if (nextProps.isFetching !== this.props.isFetching || nextProps.isError !== this.props.isError) {
      const isOpen: boolean = nextProps.isFetching || nextProps.isError;
      this.props.toggleModal(isOpen);
    }
  }
  onSubmit(data: LoginFormData) {
    this.props.submitLogin(data, (data) => this.submitCb(data));
  }

  submitCb(data: any): void {
    const {changeStep, setToken, history} = this.props;
    if (data && data.paidState !== 0) {
      setToken(data.authToken);
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      storage.save(
        'userProgram', data.program, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10}
      );
      if (data.paidState === 2) {
        changeStep(Steps.four);
      } else if (data.isFirstEdit) {
        history.replace('/profile');
      } else {
        history.replace('/task');
      }
    } else if (!data.program) {
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.two);
      history.replace('/signup');
    } else {
      storage.save('token', data.authToken, {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.three);
      history.replace('/signup');
    }
  }

  setError(){
    this.props.setError(false)
  }

  render() {
    const {isFetching, errMsg} = this.props;
    return (
      <LoginForm onSubmit={this.onSubmit.bind(this)}>
        <CustomModal
          modal={ModalTypes.error}
          resultText={errMsg}
          isLoader={isFetching}
          clickHandler={this.setError.bind(this)}
        />
      </LoginForm>
    );
  }
}

export const Login = connect(mapStateToProps, {
  submitLogin,
  changeStep,
  setToken,
  setError,
  toggleModal
})(LoginComponent);
export default LayoutHOC()(withRouter(Login));