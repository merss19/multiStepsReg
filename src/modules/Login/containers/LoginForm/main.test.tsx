import {shallow, mount} from 'enzyme'
import {LoginForm} from './main'
//import {Loader} from '../../components/Loader'
import {Field, FormProps, reduxForm} from 'redux-form/immutable';
import {Button, ButtonTypes} from '../../../../components/Button';
import * as React from 'react'
import toJson from 'enzyme-to-json'
import * as sinon from 'sinon'

const {JSDOM} = require('jsdom');
const dom = new JSDOM(`
<!doctype html>
<html>
  <body>
    <script type="text/javascript" src="https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.8&appId=602675109923486"></script>
      <script>
      !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
      n.callMethod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
      n.push=n;n.loaded=!0;n.version='2.0';n.queue=[];t=b.createElement(e);t.async=!0;
      t.src=v;s=b.getElementsByTagName(e)[0];s.parentNode.insertBefore(t,s)}(window,
      document,'script','https://connect.facebook.net/en_US/fbevents.js');
      fbq('init', '1443501655955659');
      // fbq.disablePushState = true;
    </script>
    <script>document.body.appendChild(document.createElement("h2"));</script>
  </body>
</html>`, {runScripts: "dangerously"});
const {window} = dom;
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

let props = {
  onSubmit: jest.fn(),
  handleSubmit: jest.fn()
};
describe('Component LoginForm', () => {
  const component = shallow(<LoginForm {...props}/>);
  it('renders component', () => {
    expect(component).toHaveLength(1)
  });
  it('renders correctly', () => {
    const tree = toJson(shallow(<LoginForm {...props}/>));
    expect(tree).toMatchSnapshot()
  });
  it('pass prop onSubmit', () => {
    expect(component.instance().props.onSubmit).toBe(props.onSubmit)
  });
  it('call handleSubmit', () => {
    component.find(Button).simulate('click');
    expect(props.handleSubmit).toHaveBeenCalledWith(props.onSubmit)
  });
});

