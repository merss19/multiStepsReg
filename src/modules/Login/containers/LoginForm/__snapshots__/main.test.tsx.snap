// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoginForm renders correctly 1`] = `
<div
  className="entry entry--min"
>
  <div
    className="entry__inner"
  >
    <div
      className="entry__header"
    >
      <h2
        className="entry__title text-center"
      >
        Вход в Личный кабинет
      </h2>
    </div>
    <form
      className="entry__box"
    >
      <Field
        component={[Function]}
        name="email"
        placeholder="Ваш e-mail"
        type="email"
      />
      <Field
        component={[Function]}
        name="password"
        placeholder="Ваш пароль"
        type="password"
      />
      <Component
        prefix="btn"
        styleBtn="secondary"
        type="submit"
        wide={true}
      >
        Войти
      </Component>
      <ul
        className="entry-nav mtb20"
      >
        <li
          className="entry-nav__item"
        >
          <Link
            replace={false}
            to="/signup"
          >
            Регистрация
          </Link>
        </li>
        <li
          className="entry-nav__item"
        >
          <Link
            replace={false}
            to="/restore"
          >
            Забыли пароль?
          </Link>
        </li>
      </ul>
    </form>
  </div>
</div>
`;
